misk/src/main/kotlin/misk/web/actions/ReadinessCheckService.kt

```diff
@@ -0,0 +1,89 @@
+package misk.web.actions
+
+import com.google.common.annotations.VisibleForTesting
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.common.util.concurrent.ServiceManager
+import misk.healthchecks.HealthCheck
+import misk.healthchecks.HealthStatus
+import misk.tasks.RepeatedTaskQueue
+import misk.tasks.Result
+import misk.tasks.Status
+import misk.web.ReadinessRefreshQueue
+import misk.web.WebConfig
+import wisp.logging.getLogger
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Singleton
+class ReadinessCheckService @Inject constructor(
+  private val config: WebConfig,
+  private val clock: Clock,
+  private val serviceManagerProvider: Provider<ServiceManager>,
+  @JvmSuppressWildcards private val healthChecks: List<HealthCheck>,
+  @ReadinessRefreshQueue private val taskQueue: RepeatedTaskQueue,
+): AbstractIdleService() {
+  var status: CachedStatus? = null
+    private set
+
+  private val isServiceActive = config.health_port >= 0
```

Comment:
Should we just always run this? Setting `health_port != -1` isn't required yet, so there could be a service that doesn't have it. In that case, will always be null and readiness probes will always fail

Reply:
I don't think so, but I'm far from certain. The `JettyService` does not create a health connector unless `webConfig.health_port >= 0`, which I think means the Readiness action is not accessible. If thats the case then running a health check every 2 seconds that is inaccessible is pure waste. We could be more explicit about this, but then existing configurations that rely on this behavior will not work.

Reply:
The readiness action is always accessible (without it the pod will never start). Setting `health_port != -1` only causes misk to create a dedicated thread pool for handling requests on that port. When `health_port == -1` the health probes still come in, and are handled by `ReadinessCheckAction`, they just use the same thread pool as all other requests

Reply:
Ok, I've removed this bit